build.gradle

```diff
@@ -0,0 +1,284 @@
+      }
+      dependencySet(group: 'ch.qos.logback', version: '1.2.3') {
+        entry 'logback-access'
+        entry 'logback-classic'
+        entry 'logback-core'
+      }
+      // Be aware that Log4j is used by Elasticsearch client
+      dependencySet(group: 'org.apache.logging.log4j', version: '2.8.2') {
+        entry 'log4j-api'
+        entry 'log4j-to-slf4j'
+        entry 'log4j-core'
+      }
+      dependency 'com.h2database:h2:1.3.176'
+      dependency 'com.microsoft.sqlserver:mssql-jdbc:6.2.2.jre8'
+      dependency 'mysql:mysql-connector-java:5.1.42'
+      dependency 'com.oracle.jdbc:ojdbc8:12.2.0.1.0'
+      dependency 'org.postgresql:postgresql:42.2.1'
+      dependency('org.codehaus.woodstox:woodstox-core-lgpl:4.4.0') {
+        exclude 'javax.xml.stream:stax-api'
+      }
+      dependency('org.codehaus.staxmate:staxmate:2.0.1') {
+        exclude 'org.codehaus.woodstox:stax2-api'
+        exclude 'stax:stax-api'
+        exclude 'org.codehaus.woodstox:woodstox-core-asl'
+      }
+      dependency('org.mockito:mockito-core:2.13.0') {
+        exclude 'org.hamcrest:hamcrest-core'
+      }
+      dependency 'org.assertj:assertj-core:3.9.0'
+      dependency 'org.assertj:assertj-guava:3.1.0'
+      dependency('org.codehaus.woodstox:stax2-api:3.1.4') {
+        exclude 'stax:stax-api'
+      }
+      dependency 'javax.servlet:javax.servlet-api:3.0.1'
+      dependency 'org.eclipse.jetty:jetty-server:8.1.12.v20130726'
+      dependency 'org.hamcrest:hamcrest-all:1.3'
+      dependency 'junit:junit:4.12'
+      dependency 'com.tngtech.java:junit-dataprovider:1.9.2'
+      dependency('org.dbunit:dbunit:2.4.5') {
+        exclude 'commons-lang:commons-lang'
+        exclude 'xerces:xmlParserAPIs'
+        exclude 'xerces:xercesImpl'
+      }
+      dependency 'org.eclipse.jetty:test-jetty-servlet:8.1.12.v20130726'
+      dependency 'org.reflections:reflections:0.9.9'
+      dependency 'com.google.code.gson:gson:2.3.1'
+      dependency('org.sonarsource:sonar-persistit:3.3.2') {
+        exclude 'commons-logging:commons-logging'
+      }
+      dependency 'com.github.kevinsawicki:http-request:5.4.1'
+      dependency 'org.codehaus.sonar.runner:sonar-runner-api:2.4'
+      dependency 'org.freemarker:freemarker:2.3.20'
+      dependency "com.google.protobuf:protobuf-java:${protobufVersion}"
+      dependency 'net.jpountz.lz4:lz4:1.3.0'
+      dependency 'org.subethamail:subethasmtp:3.1.7'
+      dependencySet(group: 'org.apache.tomcat.embed', version: '8.5.23') {
+        entry 'tomcat-embed-core'
+        entry('tomcat-embed-jasper') {
+          exclude 'org.eclipse.jdt.core.compiler:ecj'
+        }
+      }
+      dependencySet(group: 'com.squareup.okhttp3', version: '3.7.0') {
+        entry 'okhttp'
+        entry 'mockwebserver'
+      }
+      // TODO provided
+      dependency 'xml-apis:xml-apis:1.4.01'
+      dependency 'org.sonarsource.orchestrator:sonar-orchestrator:3.16.0.1420'
+    }
+  }
+
+  apply plugin: 'java'
```

Comment:
some subprojects are not java. An option is to group the configuration specific to java subprojects like at https://github.com/Netflix/metacat/blob/master/build.gradle#L74